frontend/src/hooks/useRpcClient.ts

```diff
@@ -0,0 +1,6 @@
+import { RpcClient } from '@/clients/rpc';
+import { JsonRpcService } from '@/services/JsonRpcService';
+
+export function useRpcClient() {
```

Comment:
@denishacquin let's do the same for viem please. We should create a /client/web3 that exposes all viem methods that we need and then a /hooks/useWeb3Client

Reply:
Is this kindof what you have in mind? useWallet would be the same as useWeb3Client I think? <img width="1700" alt="image" src="https://github.com/user-attachments/assets/66fa6c75-27ed-43f5-8e3f-4143434e6501">

Reply:
I split the client and the hook, see changes